packages/babel-parser/src/parser/expression.js

```diff
@@ -931,8 +933,8 @@ export default class ExpressionParser extends LValParser {
       elts.push(
         this.parseExprListItem(
           false,
-          possibleAsyncArrow ? new ExpressionErrors() : undefined,
-          possibleAsyncArrow ? { start: 0 } : undefined,
+          refExpressionErrors || new ExpressionErrors(),
```

Comment:
```suggestion
          refExpressionErrors,
```
I don't think we should create ExpressionErrors inside `parseCallExpressionArguments`. The ExpressionErrors should be created only when the argument is ambiguous, which means it can either be an Object Pattern in Async Arrow, or an Object Expression in Async Call function.

For example, if we always create `ExpresionErrors`, Babel will consider duplicate proto in call expression arguments are always ambiguous: For example, in this PR Babel does not throw for invalid `async?.({ __proto__: x, __proto__: y })` when the Flow plugin is enabled: [REPL](https://babeljs.io/repl/build/46638/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=IYZwngdgxg_AdACgN4AID6aAOAnA9gF1wwC4UAPAGnSz0JJTBQF8BKIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2%2Cflow&prettier=false&targets=&version=7.14.4%2Bpr.13410&externalPlugins=).

Reply:
Thanks 😄 It's probably what @nicolo-ribaudo try to explain me in his first commit but I got it only now 😆 . Sorry for that.